core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -646,7 +647,47 @@ private[spark] class AppStatusListener(
   }
 
   override def onUnpersistRDD(event: SparkListenerUnpersistRDD): Unit = {
-    liveRDDs.remove(event.rddId)
+    liveRDDs.remove(event.rddId).foreach { liveRDD =>
+      val executorsToUpdate = new HashSet[LiveExecutor]()
+      val storageLevel = liveRDD.info.storageLevel
+      val distributions = liveRDD.getDistributions()
+
+      // Use RDD distribution to update executor memory and disk usage info.
+      distributions.foreach { case (executorId, rddDist) =>
+        val maybeExec = liveExecutors.get(executorId)
```

Comment:
You don't need the variable, right? `liveExecutors.get(executorId).foreach` is enough?

(Same with `distributions` and `partitions` and `executors`.)

Reply:
Yeah, right.